### PR TITLE
Order Types and Close Accounts SDK changes

### PR DESCRIPTION
**Pull request by kelvinlau20100**

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -651,7 +851,111 @@ export class Client {
...
+        throw Error("User has no open orders account for this market!");
+      }
+
+      const [vaultOwner, _vaultSignerNonce] =
+        await utils.getSerumVaultOwnerAndNonce(
+          market,
+          DEX_PID[Exchange.network]
+        );
+
+      settleIxs.push(
```
[the remark this answers is not in the document]
Comment (kelvinlau20100) [reply src/client.ts]: updated

File: src/client.ts [hunk 2]
```diff
@@ -626,6 +776,55 @@ export class Client {
...
+        newOrderSide,
+        newOrderType,
+        newOrderClientOrderId,
+        this.marginAccountAddress,
+        this.publicKey,
+        this._openOrdersAccounts[marketIndex],
+        this._whitelistTradingFeesAddress
+      )
+    );
+    ixs.forEach((ix) => tx.add(ix));
```
Comment (xjcaa) [inline src/client.ts:824]: should we remove `forEach`?

File: src/client.ts [hunk 3]
```diff
@@ -580,6 +679,57 @@ export class Client {
...
+      cancelOrderIx(
+        marketIndex,
+        this.publicKey,
+        this._marginAccountAddress,
+        this._openOrdersAccounts[marketIndex],
+        orderId,
+        cancelSide
+      )
+    );
+    ixs.push(
```
Comment (xjcaa) [inline src/client.ts:715]: can't you just add directly to the tx.

File: src/client.ts (hunk 2 above)
Comment (xjcaa) [reply src/client.ts:824]: there's only two ix's y ou can just add directly.

**Review by xjcaa — APPROVED** [review]

**Pull request merged**